Dockerfile

```diff
@@ -0,0 +1,40 @@
+FROM ubuntu:20.04
```

Comment:
There is no good reason. The main reason why I wrote a custom Docker image was that I attempted to compile the codebase back in November. Since the steps outlined in the README seemed deprecated, I spent a great deal of time attempting to setup a development environment and decided to record the steps in a Dockerfile such that it would be easier for others to reproduce in the future.

I was unaware of the Docker image you refer to until I examined the GHA files in your PR yesterday.

Reply:
I've tried the official Docker image now, and added it to the documentation as well. One difference to the one included in this repo is that it does not seem to have the `qemu` software. The Docker image supplied here is supposed to be a one-stop for both building and testing. When I use the official image, I cannot install pip requirements specified in `tests/functional/usage.md` and hence cannot run the tests from within the image. Do you know if this is possible to do and what it would take?

Reply:
Before spending too much time on replying, let me just have a look at the functional tests pipeline. I overlooked that when I first looked at the GHA.

Reply:
I'm afraid we don't have a clean docker image for that yet. we probably need to create one at some point and to update the documentation about the developing environment set up. It shouldn't be such a nightmare.

Reply:
The Docker image in this repo works for the tests now for the `nanosp` device at least. However, I realise that I need to install the `nanosp` SDK to also be able to build within the repo, so for now I've used the official Ledger image to do that.